File: 1. cpp/1. basics/lecture 2/p6.cpp

#include <iostream>
using namespace std;
class solution {
public:
    solution(int n) {
      for(int i=n; i>=1; i--) {
        for(int j=1; j<=i; j++) {
          cout<<j<<" ";
        }
        cout<<endl;
      }
    }
};

/* pattern output:

1 2 3 4
1 2 3
1 2
1

*/

int main() {
  int size;
  cout<<"Enter size of pattern:"<<endl;
  cin>>size;

  cout<<"6. output:"<<endl;
  solution pattern(size);

}

File: 2. python/1. basics/lecture 2/p6.py
class Solution:
    def __init__(self, n): #constructor
        for i in range(n, 0, -1):
            for j in range(1,i+1):
                print(j, end=" ")
            print()

''' pattern output:

1 2 3 4
1 2 3
1 2
1

'''

if __name__ == "__main__":
    size = int(input("enter size: "))

    print("6. output:")
    Solution(size)